src/Colopl/Spanner/Concerns/ManagesTransactions.php

```diff
@@ -44,29 +45,31 @@ trait ManagesTransactions
      */
     public function transaction(Closure $callback, $attempts = Database::MAX_RETRIES)
     {
-        // Since Cloud Spanner does not support nested transactions,
-        // we use Laravel's transaction management for nested transactions only.
-        if ($this->transactions > 0) {
-            return parent::transaction($callback, $attempts);
-        }
-
-        $return = $this->getSpannerDatabase()->runTransaction(function (Transaction $tx) use ($callback) {
-            try {
-                $this->currentTransaction = $tx;
-                $this->transactions++;
-                $this->fireConnectionEvent('beganTransaction');
-                $result = $callback($this);
-                $this->performSpannerCommit();
-                return $result;
-            } catch (Throwable $e) {
-                $this->rollBack();
-                throw $e;
+		return $this->sessionNotFoundWrapper(function () use ($callback, $attempts) {
+            // Since Cloud Spanner does not support nested transactions,
+            // we use Laravel's transaction management for nested transactions only.
+            if ($this->transactions > 0) {
+                return parent::transaction($callback, $attempts);
             }
-        }, ['maxRetries' => $attempts - 1]);
-
-        $this->fireConnectionEvent('committed');
 
-        return $return;
+			$return = $this->getSpannerDatabase()->runTransaction(function (Transaction $tx) use ($callback) {
+				try {
+					$this->currentTransaction = $tx;
+					$this->transactions++;
+					$this->fireConnectionEvent('beganTransaction');
+					$result = $callback($this);
+					$this->performSpannerCommit();
+					return $result;
+				} catch (Throwable $e) {
+					$this->rollBack();
+					throw $e;
+				}
+			}, ['maxRetries' => $attempts - 1]);
+
+			$this->fireConnectionEvent('committed');
+
+			return $return;
+		});
```

Comment:
Please fix the tabs here to spaces as well.

Reply:
Sorry for the ambiguity. The entire fix here is indented with tabs. Please replace the tabs with 4 spaces.

Reply:
done